storage/innobase/dict/dict0load.cc

```diff
@@ -2516,7 +2517,7 @@ static dict_table_t *dict_load_table_one(const span<const char> &name,
 		/* Don't attempt to load the indexes from disk. */
 	} else if (err == DB_SUCCESS) {
 		err = dict_load_foreigns(table->name.m_name, nullptr,
-					 0, true, ignore_err, fk_tables);
+					 0, true, ignore_err, fk_list);
 
 		if (err != DB_SUCCESS) {
 			ib::warn() << "Load table " << table->name
```

Comment:
Instead of maintaining a local `std::deque`, could we simply save an iterator on `fk_tables` ? Something like the following?
```c++
auto i = fk_tables.end();
err = dict_load_foreigns(...);
if (err != DB_SUCCESS) {
	fk_tables.erase(i, fk_ tables.end());
```